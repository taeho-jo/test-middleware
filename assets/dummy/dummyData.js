export const data = [
  {
    label: '가',
    first: 4000,
    second: 2400,
    third: 2400,
  },
  {
    label: '나',
    first: 3000,
    second: 1398,
    third: 2210,
  },
  {
    label: '다',
    first: 2000,
    second: 9800,
    third: 2290,
  },
  {
    label: '라',
    first: 2780,
    second: 3908,
    third: 2000,
  },
  {
    label: '마',
    first: 1890,
    second: 4800,
    third: 2181,
  },
  {
    label: '바',
    first: 2390,
    second: 3800,
    third: 2500,
  },
  {
    label: '사',
    first: 3490,
    second: 4300,
    third: 2100,
  },
];
