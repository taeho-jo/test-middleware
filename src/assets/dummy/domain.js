export const DOMAIN = 'http://localhost:3000';

export const DOMAIN_ARR = [
  `${DOMAIN}/`,
  `${DOMAIN}/graph`,
  `${DOMAIN}/form`,
  `${DOMAIN}/note`,
];

// export default DOMAIN_ARR;
