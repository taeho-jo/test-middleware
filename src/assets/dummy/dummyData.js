export const verticalData = [
  {
    name: '가나다라마바사 아자차카 타파하 ABCD EFG',
    value: 30,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
  {
    name: '가나다라마바사 아자차카 타파하 ABCD EFG',
    value: 80,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
  {
    name: './src/components/atoms/SimpleBarChart/index.tsx',
    value: 90,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
  {
    name: '1234 567 89 10 기니디리 미비시 1234 567 89 10 기니디리 미비시',
    value: 100,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
  {
    name: 'YAML 또한 JSON과 비슷하게 사람이 읽기 쉬운 형태의 데이터 표현 형식입니다',
    value: 50,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
  {
    name: 'YAML',
    value: 80,
    contents:
      "Specify the domain of axis when the axis is a number axis. The length of domain should be 2, and we will validate the values in domain. And each element in the array can be a number, 'auto', 'dataMin', 'dataMax', a string like 'dataMin - 20', 'dataMax + 100', or a function that accepts a single argument and returns a number. If any element of domain is set to be 'auto', comprehensible scale ticks will be calculated, and the final domain of axis is generated by the ticks.",
  },
];

export const data = [
  {
    label: '가',
    first: 4000,
    second: 2400,
    third: 2400,
  },
  {
    label: '나',
    first: 3000,
    second: 1398,
    third: 2210,
  },
  {
    label: '다',
    first: 2000,
    second: 9800,
    third: 2290,
  },
  {
    label: '라',
    first: 2780,
    second: 3908,
    third: 2000,
  },
  {
    label: '마',
    first: 1890,
    second: 4800,
    third: 2181,
  },
  {
    label: '바',
    first: 2390,
    second: 3800,
    third: 2500,
  },
  {
    label: '사',
    first: 3490,
    second: 4300,
    third: 2100,
  },
];
