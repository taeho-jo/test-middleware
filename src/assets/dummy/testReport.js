export const reportData = {
  answerTotalCount: '69',
  eightySamplingError: '±6.16%p',
  ninetySamplingError: '±9.44%p',
  answerInfoSection: {
    genderInfoList: [
      { name: '남', value: '80', count: '250' },
      { name: '여', value: '20', count: '30' },
    ],
    ageGradeInfoList: [
      { name: '10', value: '20', count: '23' },
      { name: '20', value: '60', count: '23' },
      { name: '30', value: '100', count: '23' },
      { name: '40', value: '120', count: '23' },
      { name: '50', value: '80', count: '23' },
      { name: '60', value: '80', count: '23' },
    ],
    deviceInfoList: [
      { name: 'Android', value: '80' },
      { name: 'IOS', value: '20' },
    ],
    cellInfoList: [
      { name: '중고차 판매자', value: '10' },
      { name: '중고차 구매자', value: '20' },
      { name: '중고차 구매 보류', value: '40' },
    ],
  },
  S1: {
    uiSummerySection: {
      missionSuccess: [
        {
          index: '0',
          name: '포인트 기프티콘으로 교환하기',
          missionSuccessRatioInfo: [
            { name: '성공', value: 70 },
            { name: '실패', value: 30 },
          ],
        },
        {
          index: '1',
          name: '보유 포인트/ 참여 테스트 확인하기',
          missionSuccessRatioInfo: [
            { name: '성공', value: 80 },
            { name: '실패', value: 20 },
          ],
        },
        {
          index: '2',
          name: '회원정보 확인하기',
          missionSuccessRatioInfo: [
            { name: '성공', value: 50 },
            { name: '실패', value: 50 },
          ],
        },
      ],
      missionFatality: [
        {
          index: '0',
          name: '포인트 기프티콘으로 교환하기',
          completeList: [
            '사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '카테고리 세분화가 필요할것같습니다.',
            '카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          completeComment: 'comment',
          // 서비스 전체 미션별 추가기능 피드백
          additionalList: [
            '사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '카테고리 세분화가 필요할것같습니다.',
            '카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          additionalComment: 'comment',
          // 서비스 전체 미션별 시스템오류 피드백
          systemErrorList: [
            '사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '카테고리 세분화가 필요할것같습니다.',
            '카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          systemErrorComment: 'comment',
          value: 30, // 언급비율
          fatality: '2.9', // 치명도
          missionFunctionFatality: [
            {
              index: '0',
              name: '알림톡으로 테스트 알림받기',
              value: 49,
              fatality: '4.3',
              info: '1전체 350개의 응답중 기능 알림톡으로 테스트 알림받기에 대한 언급빈도는 168개이며, 4.35(많이 치명적임)의 치명도가 나왔습니다.구체적으로 어떤 부분이 불편했는지는 다음 그래프에서 상세하게 확인하실 수 있습니다.',
              detailInfo: '기능에 A에 대해 언급 된.....',
              detailList: [
                {
                  name: '컴포넌트(조회 콘텐츠)',
                  stack: [
                    {
                      value: 20,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 40,
                      reason: '불필요한 정보가 많다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
                {
                  name: '컴포넌트(메뉴)',
                  stack: [
                    {
                      value: 30,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 55,
                      reason: '가독성이 낮다.2222',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
              ],
              usabilityList: [
                { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: 46, fatality: '4.35', mention: '43.5' },
                { name: '유저의 선택권 및 자유도', value: 18, fatality: '2.3', mention: '23' },
                { name: '시스템과 살제 세상의 매칭', value: 58, fatality: '5.3', mention: '53' },
              ],
            },
            {
              index: '1',
              name: '알림톡으로 테스트 알림받기2',
              value: 49,
              fatality: '4.3',
              info: '2전체 350개의 응답중 기능 알림톡으로 테스트 알림받기에 대한 언급빈도는 168개이며, 4.35(많이 치명적임)의 치명도가 나왔습니다.구체적으로 어떤 부분이 불편했는지는 다음 그래프에서 상세하게 확인하실 수 있습니다.',
              detailInfo: '기능에 A에 대해 언급 된.....',
              detailList: [
                {
                  name: '컴포넌트(조회 콘텐츠)',
                  stack: [
                    {
                      value: 60,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 10,
                      reason: '불필요한 정보가 많다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
                {
                  name: '컴포넌트(메뉴)',
                  stack: [
                    {
                      value: 40,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 20,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
              ],
              usabilityList: [
                { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: 46, fatality: '4.35', mention: '43.5' },
                { name: '유저의 선택권 및 자유도', value: 18, fatality: '2.3', mention: '23' },
                { name: '시스템과 살제 세상의 매칭', value: 58, fatality: '5.3', mention: '53' },
              ],
            },
          ],
        },
        {
          index: '1',
          name: '보유 포인트/ 참여 테스트 확인하기',
          completeList: [
            '22사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '22카테고리 세분화가 필요할것같습니다.',
            '22카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          completeComment: '22comment',
          // 서비스 전체 미션별 추가기능 피드백
          additionalList: [
            '22사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '22카테고리 세분화가 필요할것같습니다.',
            '22카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          additionalComment: '22comment',
          // 서비스 전체 미션별 시스템오류 피드백
          systemErrorList: [
            '22사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '22카테고리 세분화가 필요할것같습니다.',
            '22카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          systemErrorComment: '22comment',
          value: 46.5,
          fatality: '3.8',
          missionFunctionFatality: [
            {
              index: '0',
              name: '알림톡으로 테스트 알림받기',
              value: 49,
              fatality: '4.3',
              info: '3전체 350개의 응답중 기능 알림톡으로 테스트 알림받기에 대한 언급빈도는 168개이며, 4.35(많이 치명적임)의 치명도가 나왔습니다.구체적으로 어떤 부분이 불편했는지는 다음 그래프에서 상세하게 확인하실 수 있습니다.',
              detailInfo: '기능에 A에 대해 언급 된.....',
              detailList: [
                {
                  name: '컴포넌트(조회 콘텐츠)',
                  stack: [
                    {
                      value: 40,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 30,
                      reason: '불필요한 정보가 많다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
                {
                  name: '컴포넌트(메뉴)',
                  stack: [
                    {
                      value: 20,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 70,
                      reason: '가독성이 낮다2222.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
              ],
              usabilityList: [
                { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: 46, fatality: '4.35', mention: '43.5' },
                { name: '유저의 선택권 및 자유도', value: 18, fatality: '2.3', mention: '23' },
                { name: '시스템과 살제 세상의 매칭', value: 58, fatality: '5.3', mention: '53' },
              ],
            },
          ],
        },
        {
          index: '2',
          name: '회원정보 확인하기',
          completeList: [
            '33사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '33카테고리 세분화가 필요할것같습니다.',
            '33카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          completeComment: '33comment',
          // 서비스 전체 미션별 추가기능 피드백
          additionalList: [
            '33사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '33카테고리 세분화가 필요할것같습니다.',
            '33카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          additionalComment: '33comment',
          // 서비스 전체 미션별 시스템오류 피드백
          systemErrorList: [
            '33사용하는 것의 불편사항보다는 건의사항 이지만 큰 카테고리(리빙) 안에 작은 카테고리로(청소/수리)등이 있으면 좋을 것 같습니다.',
            '33카테고리 세분화가 필요할것같습니다.',
            '33카테고리가 모호해서 광범위해서 지정하기가 좀 어려웠어요.',
          ],
          systemErrorComment: '33comment',
          value: 61.8,
          fatality: '4.35',
          missionFunctionFatality: [
            {
              index: '0',
              name: '알림톡으로 테스트 알림받기',
              value: 49,
              fatality: '4.3',
              info: '4전체 350개의 응답중 기능 알림톡으로 테스트 알림받기에 대한 언급빈도는 168개이며, 4.35(많이 치명적임)의 치명도가 나왔습니다.구체적으로 어떤 부분이 불편했는지는 다음 그래프에서 상세하게 확인하실 수 있습니다.',
              detailInfo: '기능에 A에 대해 언급 된.....',
              detailList: [
                {
                  name: '컴포넌트(조회 콘텐츠)',
                  stack: [
                    {
                      value: 10,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 10,
                      reason: '불필요한 정보가 많다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
                {
                  name: '컴포넌트(메뉴)',
                  stack: [
                    {
                      value: 10,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                    {
                      value: 10,
                      reason: '가독성이 낮다.',
                      answer: ['주관식 내용1', '주관식 내용2', '주관식 내용3', '주관식 내용4'],
                    },
                  ],
                  count: '49',
                  rate: '49.5',
                },
              ],
              usabilityList: [
                { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: 46, fatality: '4.35', mention: '43.5' },
                { name: '유저의 선택권 및 자유도', value: 18, fatality: '2.3', mention: '23' },
                { name: '시스템과 살제 세상의 매칭', value: 58, fatality: '5.3', mention: '53' },
              ],
            },
          ],
        },
      ],
      serviceTotalUsabilityList: [
        { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: '46', fatality: '4.35', mention: '43.5' },
        { name: '유저의 선택권 및 자유도', value: '18', fatality: '2.3', mention: '23' },
        { name: '시스템과 살제 세상의 매칭', value: '58', fatality: '5.3', mention: '53' },
        { name: '유저가 에러를 인지/판단/복구하도록 돕기', value: '46', fatality: '4.35', mention: '43.5' },
        { name: '유저의 선택권 및 자유도', value: '18', fatality: '2.3', mention: '23' },
        { name: '시스템과 살제 세상의 매칭', value: '58', fatality: '5.3', mention: '53' },
      ],
    },
  },
};
