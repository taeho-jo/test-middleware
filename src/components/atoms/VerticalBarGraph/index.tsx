import React, { useCallback, useState } from 'react';
import {
  Bar,
  BarChart,
  CartesianGrid,
  LabelList,
  Legend,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis,
  Cell,
} from 'recharts';

// Types
interface DummyDataType {
  [key: string]: string | number | object;
}
interface PropsType {
  data: DummyDataType[];
  barColor?: string[];
  cartesianGrid?: boolean;
  xAxis?: boolean;
  yAxis?: boolean;
  tooltip?: boolean;
  legend?: boolean;
}

const VerticlBarGraph = ({
  cartesianGrid = true,
  xAxis = true,
  yAxis = true,
  tooltip = true,
  legend = true,
  barColor = ['#067373', '#FEA443', '#812F33'],
  data,
}: PropsType) => {
  const [focusBar, setFocusBar] = useState<number | null>(null);
  const [hoverBar, setHoverBar] = useState<number | null>(null);

  const renderCustomizedLabel = useCallback(props => {
    const { x, y, height, value } = props;
    return (
      <text
        fontSize={14}
        x={x + 20}
        y={y + height / 2 + 6}
        fill={'#0D0D0D'}
        fontWeight={'bold'}
        textAnchor="start"
      >
        {value}
      </text>
    );
  }, []);

  const handleSelectBarChart = useCallback(
    state => {
      console.log(state);
      if (state) {
        setFocusBar(state.activeTooltipIndex);
      } else {
        setFocusBar(null);
      }
    },
    [focusBar],
  );

  const handleHoverBarChart = useCallback(
    state => {
      if (state) {
        setHoverBar(state.activeTooltipIndex);
      } else {
        setHoverBar(null);
      }
    },
    [hoverBar],
  );

  return (
    <ResponsiveContainer width={'100%'} height={'100%'}>
      <BarChart
        data={data}
        margin={{
          top: 5,
          // right: 30,
          // left: 20,
          bottom: 5,
        }}
        layout="vertical"
        onClick={state => handleSelectBarChart(state)}
        onMouseMove={state => handleHoverBarChart(state)}
      >
        {cartesianGrid ? <CartesianGrid strokeDasharray="2 2" /> : null}
        {xAxis ? <XAxis type={'number'} domain={[0, 100]} /> : null}
        {yAxis ? <YAxis type="category" hide /> : null}
        {tooltip ? <Tooltip cursor={false} /> : null}
        {legend ? <Legend /> : null}
        <Bar
          background={{ fill: '#F2F2F2' }}
          dataKey={'value'}
          fill={'#FEA443'}
        >
          {data.map((entry, index) => (
            <Cell
              key={index}
              fill={
                focusBar === index
                  ? '#C8D9C7'
                  : hoverBar === index
                  ? '#D9B95B'
                  : '#F2DB94'
              }
            />
          ))}
          <LabelList
            dataKey={'name'}
            position="insideLeft"
            style={{ fill: '#fff', fontWeight: 'bold' }}
            content={renderCustomizedLabel}
          />
        </Bar>
      </BarChart>
    </ResponsiveContainer>
  );
};

export default VerticlBarGraph;
