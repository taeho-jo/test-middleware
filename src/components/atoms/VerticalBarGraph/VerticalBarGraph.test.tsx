import React from 'react';
import { render } from '@testing-library/react';
import VerticalBarGraph from './index';
import { verticalData } from '../../../assets/dummy/dummyData';

describe('Vertical Bar Graph', () => {
  it('vertical bar graph', () => {
    const utils = render(<VerticalBarGraph data={verticalData} />);
    expect(utils.container).toMatchSnapshot();
  });
});
