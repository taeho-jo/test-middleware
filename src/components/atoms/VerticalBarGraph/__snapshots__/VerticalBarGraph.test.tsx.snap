// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Bar Graph vertical bar graph 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 100%;"
  />
</div>
`;
